src/libraries/System.Text.Json/tests/Serialization/CustomConverterTests/CustomConverterTests.StructMember.cs

```diff
@@ -0,0 +1,99 @@
+
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class CustomConverterTests
+    {
+        private class StructToInterfaceConverter : JsonConverter<IMemberInterface>
+        {
+            public override bool CanConvert(Type typeToConvert)
+            {
+                return typeof(IMemberInterface).IsAssignableFrom(typeToConvert);
+            }
+
+            public override IMemberInterface Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                string value = reader.GetString();
+
+                return new StructMember(value);
+            }
+
+            public override void Write(Utf8JsonWriter writer, IMemberInterface value, JsonSerializerOptions options)
+            {
+                JsonSerializer.Serialize(writer, value.Value, typeof(string), options);
+            }
+        }
+
+        [Fact]
+        public static void StructMemberToInterfaceConverter()
+        {
+            var options = new JsonSerializerOptions()
+            {
+                IncludeFields = true,
+            };
+            options.Converters.Add(new StructToInterfaceConverter());
+
+            string json;
+
+            {
+                TestClassWithStructMember obj = new TestClassWithStructMember();
+                obj.Initialize();
+                obj.Verify();
+                json = JsonSerializer.Serialize(obj, options);
+            }
+
+            {
+                TestClassWithStructMember obj = JsonSerializer.Deserialize<TestClassWithStructMember>(json, options);
+                obj.Verify();
+            }
+        }
+
+        private class StructToObjectConverter : JsonConverter<object>
+        {
+            public override bool CanConvert(Type typeToConvert)
+            {
+                return typeof(IMemberInterface).IsAssignableFrom(typeToConvert);
+            }
+
+            public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                string value = reader.GetString();
+
+                return new StructMember(value);
+            }
+
+            public override void Write(Utf8JsonWriter writer, object value, JsonSerializerOptions options)
+            {
+                JsonSerializer.Serialize(writer, ((IMemberInterface)value).Value, typeof(string), options);
+            }
+        }
+
+        [Fact]
+        public static void StructMemberToObjectConverter()
+        {
+            var options = new JsonSerializerOptions()
+            {
+                IncludeFields = true,
+            };
+            options.Converters.Add(new StructToObjectConverter());
+
+            string json;
+
+            {
+                TestClassWithStructMember obj = new TestClassWithStructMember();
+                obj.Initialize();
+                obj.Verify();
+                json = JsonSerializer.Serialize(obj, options);
```

Comment:
Please verify the result of the serialization and ensure that the converter's write method was called.